roles/uperf/tasks/setup.yml

```
@@ -0,0 +1,167 @@
+---
+
+- name: Get current state
+  k8s_facts:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: '{{ meta.name }}'
+    namespace: '{{ operator_namespace }}'
+  register: resource_state
+
+- operator_sdk.util.k8s_status:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: "{{ meta.name }}"
+    namespace: "{{ operator_namespace }}"
+    status:
+      state: Building
+      complete: false
+  when: resource_state.resources[0].status.state is not defined
+
+- name: Get current state - If it has changed
+  k8s_facts:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: '{{ meta.name }}'
+    namespace: '{{ operator_namespace }}'
+  register: resource_state
+
+- name: Capture operator information
+  k8s_facts:
+    kind: Pod
+    api_version: v1
+    namespace: '{{ operator_namespace }}'
+    label_selectors:
+      - name = benchmark-operator
+  register: bo
+  
+- block:
+    #
+    # This block is for scale mode where client and server pods are spreaded
+    # across all eligible nodes
+    #
+  - name: List Nodes Labeled as Workers
+    k8s_info:
+      api_version: v1
+      kind: Node
+      label_selectors:
+        - "node-role.kubernetes.io/worker="
+    register: node_list
+    no_log: True
+  
+  - name: Isolate Worker Role Hostnames
+    set_fact:
+      worker_node_list: "{{ node_list | json_query('resources[].metadata.name') | list }}"
```

As we talked internally I think this expression would fit better and prevent some of the issues the CI has.

```suggestion
      worker_node_list: "{{ node_list | json_query('resources[].metadata.metadata.labels.kubernetes\.io/hostname') | list }}"
```

---

Yes. Let me do it